tests/pocket_test.cpp

```diff
@@ -0,0 +1,832 @@
+    REQUIRE( data_jug.volume_capacity > 5 * rag.volume() );
+    REQUIRE( data_jug.volume_capacity > 10 * screwdriver.volume() );
+
+    // But it has a narrow mouth, just barely enough to fit a screwdriver
+    data_jug.max_item_volume = screwdriver.volume();
+    // And the rock is definitely too big to fit
+    REQUIRE( rock.volume() > 2 * screwdriver.volume() );
+
+    GIVEN( "large watertight pocket with a narrow opening" ) {
+        item_pocket pocket_jug( &data_jug );
+
+        THEN( "it can contain liquids" ) {
+            expect_can_contain( pocket_jug, water );
+        }
+        THEN( "it can contain solid items that fit through the opening" ) {
+            expect_can_contain( pocket_jug, screwdriver );
+        }
+        THEN( "it can contain soft items larger than the opening" ) {
+            expect_can_contain( pocket_jug, rag );
+        }
+        THEN( "it cannot contain solid items larger than the opening" ) {
+            expect_cannot_contain( pocket_jug, rock, "item too big",
+                                   item_pocket::contain_code::ERR_TOO_BIG );
+        }
+    }
+}
+
+// Max container volume
+// --------------------
+// Normally, the "max_contains_volume" from pocket data JSON is the largest total volume of
+// items the pocket can hold. It is returned by the pocket_data::max_contains_volume() function.
+//
+// However, pockets with an "ammo_restriction", the pocket_data::max_contains_volume() is derived
+// from the size of the ammo given in the restriction.
+//
+// Related JSON fields:
+// "max_contains_volume"
+// "ammo_restriction"
+//
+// Functions:
+// pocket_data::max_contains_volume
+//
+TEST_CASE( "max container volume", "[pocket][max_contains_volume]" )
+{
+    // TODO: Add tests for having multiple ammo types in the ammo_restriction
+
+    WHEN( "pocket has no ammo_restriction" ) {
+        pocket_data data_box( item_pocket::pocket_type::CONTAINER );
+        // Just a normal 1-liter box
+        data_box.volume_capacity = 1_liter;
+        REQUIRE( data_box.ammo_restriction.empty() );
+
+        THEN( "max_contains_volume is the pocket's volume_capacity" ) {
+            CHECK( data_box.max_contains_volume() == 1_liter );
+        }
+    }
+
+    WHEN( "pocket has ammo_restriction" ) {
+        pocket_data data_ammo_box( item_pocket::pocket_type::CONTAINER );
+
+        // 9mm ammo is 50 rounds per 250ml (or 200 rounds per liter), so this ammo box
+        // should be exactly 1 liter in size, so it can contain this much ammo.
+        data_ammo_box.ammo_restriction.emplace( ammotype( "9mm" ), 200 );
+        REQUIRE_FALSE( data_ammo_box.ammo_restriction.empty() );
+
+        // And because actual volume is derived from ammo needs, this volume should be ignored.
+        data_ammo_box.volume_capacity = 1_ml;
+
+        THEN( "max_contains_volume is based on the pocket's ammo type" ) {
+            CHECK( data_ammo_box.max_contains_volume() == 1000_ml );
```

Comment:
```suggestion
            CHECK( data_ammo_box.max_contains_volume() == 1_liter );
```
any reason this isn't just 1 liter?